For null, bigint Query fails with error "inconsistent type deduced for parameter" for null and bigint

Tested with version 0.2.8

### Passing null to timestamp column fails.

```javascript
   pg.sql `create table mb( id bigint, updated timestamp null, primary key(id) )`;

   const mb = { id: 1, updated: null };

   pg.query ("insert into mb(id, updated) values ($1, $2)", mb.id, mb.updated) ;
```

### Passing 1 to bigint column fails.

```javascript
   pg.sql `create table mb( id bigint, updated bigint null, primary key(id) )`;

   const mb = { id: 1, updated: 1 };

   pg.query ("insert into mb(id, updated) values ($1, $2)", mb.id, mb.updated) ;
```

Error Detail,

```
index.ts:586 error: inconsistent types deduced for parameter $2
    at me.Ce (https://cdn.jsdelivr.net/npm/@electric-sql/pglite@0.2.8/dist/chunk-AZTFRW3V.js:1:12465)
    at me.je (https://cdn.jsdelivr.net/npm/@electric-sql/pglite@0.2.8/dist/chunk-AZTFRW3V.js:1:9851)
    at me.parse (https://cdn.jsdelivr.net/npm/@electric-sql/pglite@0.2.8/dist/chunk-AZTFRW3V.js:1:8603)
    at H.execProtocol (https://cdn.jsdelivr.net/npm/@electric-sql/pglite@0.2.8/dist/index.js:1:6241)
    at async Object.execProtocol (https://cdn.jsdelivr.net/npm/@electric-sql/pglite@0.2.8/dist/worker/index.js:1:6990)
    at async BroadcastChannel.<anonymous> (https://cdn.jsdelivr.net/npm/@electric-sql/pglite@0.2.8/dist/worker/index.js:1:6564)
```

--- a/.changeset/fuzzy-books-promise.md
+++ b/.changeset/fuzzy-books-promise.md
@@ -1,0 +1,5 @@
+---
+'@electric-sql/pglite': patch
+---
+
+Improve type serialization so it matches exceptions from other libraries

--- a/packages/pglite/src/types.ts
+++ b/packages/pglite/src/types.ts
@@ -73,9 +73,17 @@ export const BOOL = 16,
 
 export const types = {
   string: {
-    to: 0,
+    to: 25,
     from: [TEXT, VARCHAR, BPCHAR],
-    serialize: (x: string) => x,
+    serialize: (x: string | number) => {
+      if (typeof x === 'string') {
+        return x
+      } else if (typeof x === 'number') {
+        return x.toString()
+      } else {
+        throw new Error('Invalid input for string type')
+      }
+    },
     parse: (x: string) => x,
   },
   number: {

--- a/packages/pglite/src/types.ts
+++ b/packages/pglite/src/types.ts
@@ -73,7 +73,7 @@ export const BOOL = 16,
 
 export const types = {
   string: {
-    to: 25,
+    to: TEXT,
     from: [TEXT, VARCHAR, BPCHAR],
     serialize: (x: string | number) => {
       if (typeof x === 'string') {

--- a/packages/pglite/tests/types.test.ts
+++ b/packages/pglite/tests/types.test.ts
@@ -81,6 +81,14 @@ describe('serialize', () => {
     expect(types.serializers[25]('test')).toEqual('test')
   })
 
+  it('string from number', () => {
+    expect(types.serializers[25](1)).toEqual('1')
+  })
+
+  it('not string', () => {
+    expect(() => types.serializers[25](true)).toThrow()
+  })
+
   it('number', () => {
     expect(types.serializers[0](1)).toEqual('1')
     expect(types.serializers[0](1.1)).toEqual('1.1')
@@ -94,10 +102,30 @@ describe('serialize', () => {
     expect(types.serializers[16](true)).toEqual('t')
   })
 
+  it('not bool', () => {
+    expect(() => types.serializers[16]('test')).toThrow()
+  })
+
   it('date', () => {
     expect(
       types.serializers[1184](new Date('2021-01-01T00:00:00.000Z')),
     ).toEqual('2021-01-01T00:00:00.000Z')
+  })
+
+  it('date from number', () => {
+    expect(types.serializers[1184](1672531200000)).toEqual(
+      '2023-01-01T00:00:00.000Z',
+    )
+  })
+
+  it('date from string', () => {
+    expect(types.serializers[1184]('2021-01-01T00:00:00.000Z')).toEqual(
+      '2021-01-01T00:00:00.000Z',
+    )
+  })
+
+  it('not date', () => {
+    expect(() => types.serializers[1184](true)).toThrow()
   })
 
   it('json', () => {
@@ -108,5 +136,9 @@ describe('serialize', () => {
     expect(types.serializers[17](Uint8Array.from([1, 2, 3]))).toEqual(
       '\\x010203',
     )
+  })
+
+  it('not blob', () => {
+    expect(() => types.serializers[17](1)).toThrow()
   })
 })

